src/pages/index.js

```diff
@@ -6,159 +7,235 @@ import PopupWithForm from "../components/PopupWithForm.js";
+}
+
+// Delete Card
+
+const deleteImageModal = document.querySelector("#delete-image-modal");
+const deleteImageConfirmationButton =
+  deleteImageModal.querySelector(".modal__button");
+
+deleteImageConfirmationButton.addEventListener("click", () => {
+  deleteConfirmation.setLoading(true);
+});
+
+const deleteConfirmation = new PopupDeleteConfirm(
+  "#delete-image-modal",
+  handleDeleteCard
+);
+deleteConfirmation.setEventListeners();
+
+function handleDeleteCard(cardData) {
+  deleteConfirmation.open();
+
+  deleteConfirmation.confirmDelete(() => {
+    api
+      .deleteCard(cardData.getCardId())
+      .then(() => {
+        cardData.removeCard();
+        deleteConfirmation.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => deleteConfirmation.setLoading(false));
+  });
 }
 
+//Liking and Disliking Cards
+
+function handleCardLike(cardData) {
+  if (!cardData._like) {
+    api.likeCard(cardData.getCardId()).then(() => {
+      cardData.handleLike(true);
+    });
+  } else if (cardData._like) {
+    api.dislikeCard(cardData.getCardId()).then(() => {
+      cardData.handleLike(false);
+    });
+  }
+}
+
+// Initial Cards
+
+const cardList = new Section(
+  {
+    renderer: (items) => {
+      cardList.addItem(items);
+    },
+  },
+  ".cards__list"
+);
+
 function getCardElement(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleCardLike
+  );
   const element = card.getView();
   return element;
 }
 
-function renderCard(cardData, wrapper) {
-  const element = getCardElement(cardData);
-  wrapper.prepend(element);
+async function cardListData() {
+  const initialCardData = await api.getInitialCards();
+  const cardElement = initialCardData.map((cardData) =>
+    getCardElement(cardData)
+  );
+  cardList.renderItems(cardElement);
 }
 
-initialCards.forEach((cardData) => renderCard(cardData, cardListEL));
+cardListData();
 
 // New Cards
+
+const newCardPopup = new PopupWithForm("#add-modal", handleAddCardSubmit);
+newCardPopup.setEventListeners();
+
 imageAddButton.addEventListener("click", () => {
   newCardPopup.open();
 });
 
-const addCardFormElement = document.querySelector("#add-card-form");
-const cardTitleInput = addCardFormElement.querySelector(
-  ".modal__input_type_title"
-);
-const cardUrlInput = addCardFormElement.querySelector(".modal__input_type_url");
-
-function handleAddCardSubmit() {
-  const name = cardTitleInput.value;
-  const link = cardUrlInput.value;
-  renderCard({ name, link }, cardListEL);
+function handleAddCardSubmit(inputValue) {
+  const name = inputValue.title;
+  const link = inputValue.link;
+  api.createNewCard({ name, link }).then((card) => {
+    const cardElement = getCardElement(card);
+    cardList.addItem(cardElement);
+  });
   newCardPopup.reset();
   newCardPopup.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well